cassandra_backup_service.py

```diff
@@ -1320,6 +1326,32 @@ def save_manifest(self, keyspace, columnfamily, manifest):
 
         return manifest_file_path
 
+    def manifest_data_retention_pruning(self, data):
+        """
+        Prune provided data to only values within the defined self.retention_days time range.
+
+        :param dict data: manifest data.
+        """
+        if 'full' in data:
+            full_to_delete = []
+            for full_ts in data['full']:
+                full_days_old = int(time.mktime(time.gmtime()) - time.mktime(time.gmtime(int(full_ts)/1000)))/86400
+                if full_days_old > self.retention_days:
+                    full_to_delete.append(full_ts)
+
+            for full_ts in full_to_delete:
+                del data['full'][full_ts]
```

Comment:
Lets add a debug line every time we throw out a file with why we are throwing it out and how old it was.